examples/tutorial_3_streamproto.md

```diff
@@ -0,0 +1,98 @@
+In the [previous tutorial](tutorial_2_customproto.md), we dug a little deeper in to how the ping protocol works by creating a custom ping protocol ourselves.
+
+This tutorial will focus on how to create a custom streaming protocol using libp2p.
+
+# Custom protocol in libp2p
+Let's create a `part3.nim`, and import our dependencies:
+```nim
+import bearssl
+import chronos
+import stew/byteutils
+
+import libp2p
+```
+This is similar to the second tutorial. Next, we'll declare our custom protocol.
+```nim
+const TestStreamCodec = "/test/stream-proto/1.0.0"
+
+type TestStreamProto = ref object of LPProtocol
+```
+
+Just as we did in the last tutorial, we've set a [protocol ID](https://docs.libp2p.io/concepts/protocols/#protocol-ids), and created a custom `LPProtocol`.
+
+As in the last tutorial, we are going to handle our client and server parts separately. The server will wait for an instruction from the client, then send a stream of data to the client and finish by closing the connection. The client will listen in a loop until the server has finished and closed the connection. Then the client will close its end of the connection as well.
+
+Lets start with the server part:
+```nim
+proc new(T: typedesc[TestStreamProto]): T =
```

Comment:
I think `type` is preferred over `typedesc`?

Reply:
```sh
$ grep -rn "typedesc" libp2p | wc -l
121
```
I guess we have to update the entire codebase

Reply:
Yeah, the current state isn't indicative of whats preferred ;)

Reply:
I personally like having brackets in the expression, because it makes it more clear that this is a generic, imo

Could we switch to `type[TestStreamProto]` instead, so that everybody is happy (well, at least me)? (across the codebase ofc)